BlocBrowser/BlocBrowser/ViewController.m

```diff
@@ -180,10 +162,59 @@ - (void) updateButtonsAndTitle {
         [self.activityIndicator stopAnimating];
     }
     
-    self.backButton.enabled = [self.webView canGoBack];
-    self.forwardButton.enabled = [self.webView canGoForward];
-    self.stopButton.enabled = self.webView.isLoading;
-    self.reloadButton.enabled = !self.webView.isLoading;
+    [self.toolbar setEnabled:[self.webView canGoBack] forButtonWithTitle:kWebBrowserBackString];
+    [self.toolbar setEnabled:[self.webView canGoForward] forButtonWithTitle:kWebBrowserForwardString];
+    [self.toolbar setEnabled:[self.webView isLoading] forButtonWithTitle:kWebBrowserStopString];
+    [self.toolbar setEnabled:(![self.webView isLoading] && self.webView.URL) forButtonWithTitle:kWebBrowserRefreshString];
+}
+
+- (void) resetWebView {
+    [self.webView removeFromSuperview];
+    
+    WKWebView *newWebView = [WKWebView new];
+    newWebView.navigationDelegate = self;
+    [self.view addSubview:newWebView];
+    
+    self.webView = newWebView;
+    
+    self.urlTextField.text = nil;
+    [self updateButtonsAndTitle];
+}
+
+#pragma mark - AwesomeFloatingToolbarDelegate
+
+- (void)floatingToolbar:(AwesomeFloatingToolbar *)toolbar didSelectButtonWithTitle:(NSString *)title {
+    if ([title isEqualToString:kWebBrowserBackString]) {
+        [self.webView goBack];
+    } else if ([title isEqualToString:kWebBrowserForwardString]) {
+        [self.webView goForward];
+    } else if ([title isEqualToString:kWebBrowserStopString]) {
+        [self.webView stopLoading];
+    } else if ([title isEqualToString:kWebBrowserRefreshString]) {
+        [self.webView reload];
+    }
+}
+
+- (void)floatingToolbar:(AwesomeFloatingToolbar *)toolbar didTryToPanWithOffset:(CGPoint)offset {
+    CGPoint startingPoint = toolbar.frame.origin;
+    CGPoint newPoint = CGPointMake(startingPoint.x + offset.x, startingPoint.y + offset.y);
+    
+    CGRect potentialNewFrame = CGRectMake(newPoint.x, newPoint.y, CGRectGetWidth(toolbar.frame), CGRectGetHeight(toolbar.frame));
+    
+    if (CGRectContainsRect(self.view.bounds, potentialNewFrame)) {
+        toolbar.frame = potentialNewFrame;
+    }
+}
+
+- (void)floatingToolbar:(AwesomeFloatingToolbar *)toolbar didTryToPinchWithScale:(CGFloat)scale {
```

Comment:
An easier way to write this code might be to do something like this:
```
    toolbar.transform = CGAffineTransformScale(toolbar.transform, scale, scale);
```
Use UIKit's built-in ability to apply transforms to views and the scale you get passed, anyway.

That code can replace all of the code in this method. If you want to retain the constraint of not having the toolbar go off the screen you can apply the transform to an off-screen view which has the same dimensions as the toolbar and then perform your test below in `CGRectContainsRect(self.view.bounds, potentialNewFrame)`. Let me know if you have questions about that approach.